simapp/test_helpers.go

```diff
@@ -77,15 +80,52 @@ func SetupWithGenesisAccounts(genAccs []authexported.GenesisAccount, balances ..
 	return app
 }
 
-// AddTestAddrs constructs and returns accNum amount of accounts with an
-// initial balance of accAmt
-func AddTestAddrs(app *SimApp, ctx sdk.Context, accNum int, accAmt sdk.Int) []sdk.AccAddress {
+type GenerateAccountStrategy func(int) []sdk.AccAddress
+
+func Random(accNum int) []sdk.AccAddress {
```

Comment:
`Random` what? Poor function name. Let's pick something more descriptive. Also add a godoc

Reply:
The idea behind that is that there was a function in simapp to create random addresses:

```func AddTestAddrs(app *SimApp, ctx sdk.Context, accNum int, accAmt sdk.Int) []sdk.AccAddress```

But this one returned some addresses in random order but a big amount of tests in the staking used a generateAddress function that returned them in incremental order. That means that the tests can call the Keeper by address and always returning the same order.

So after debugging I decided that given there was a big amount of tests that would need to be changed I created a variation of the generation of AddTestAddrs.

AddTestAddrs (original one without changing signature, I decided this if someone used this and expected that it worked the same) and AddTestAddrsIncremental (returns addresses in incremental order)

Both call internally ```func addTestAddrs(app *SimApp, ctx sdk.Context, accNum int, accAmt sdk.Int, strategy GenerateAccountStrategy) []sdk.AccAddress```  and this last param is the strategy they use.

```type GenerateAccountStrategy func(int) []sdk.AccAddress```

That is the Incremental and Random functions you saw. I can make them private to the package but a longer descriptive name maybe does not bring any value, maybe randomStrategy and incrementalStrategy?